src/tempstorage/TemporaryStorage.py

```diff
@@ -84,6 +85,14 @@ def __init__(self, name='TemporaryStorage'):
 
         _conflict_cache_maxage -- age at whic conflict cache items are GC'ed
         """
+        deprecation_warning = """\
+DEPRECATED: Usage of the package tempstorage is deprecated, as it is known to randomly lose data.
+Especially on zope 4. For details see https://github.com/zopefoundation/tempstorage/issues/8
+and https://github.com/zopefoundation/tempstorage
+"""
+        LOG.warning(deprecation_warning)
+        warnings.warn(deprecation_warning, DeprecationWarning)
```

Comment:
Do we need both logging and warnings?

Reply:
The warning is visible in the log by default, while the warning is not, but is on the other hand searchable with tooling. Thus both - but I am not familiar with how this is handled in other parts of zope, so I am open to suggestions.

Reply:
There are many ways, e. g. [zope.deprecation](https://pypi.org/project/zope.deprecation/).

Reply:
That module seems to only call warnings.warn(..., DeprecationWarning) while being much more complex and adding a dependency just for this.

Knowing that, I would prefer to have this simple solution that also guarantees that people get the log entry.

Reply:
@icemac Are you ok with that? If so, please merge or indicate consent so I can do that.